src/Microsoft.Data/PrimitiveDataFrameColumnArithmetic.cs

```diff
@@ -0,0 +1,4138 @@
+       void LessThan(PrimitiveDataFrameColumnContainer<T> column, T scalar, PrimitiveDataFrameColumnContainer<bool> ret);
+    }
+
+    internal static class PrimitiveDataFrameColumnArithmetic<T>
+        where T : struct
+    {
+        public static IPrimitiveDataFrameColumnArithmetic<T> Instance => PrimitiveDataFrameColumnArithmetic.GetArithmetic<T>();
+    }
+
+    internal static class PrimitiveDataFrameColumnArithmetic
+    {
+        public static IPrimitiveDataFrameColumnArithmetic<T> GetArithmetic<T>()
+            where T : struct
+        {
+            if (typeof(T) == typeof(bool))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new BoolArithmetic();
+            }
+            else if (typeof(T) == typeof(byte))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new ByteArithmetic();
+            }
+            else if (typeof(T) == typeof(char))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new CharArithmetic();
+            }
+            else if (typeof(T) == typeof(decimal))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new DecimalArithmetic();
+            }
+            else if (typeof(T) == typeof(double))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new DoubleArithmetic();
+            }
+            else if (typeof(T) == typeof(float))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new FloatArithmetic();
+            }
+            else if (typeof(T) == typeof(int))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new IntArithmetic();
+            }
+            else if (typeof(T) == typeof(long))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new LongArithmetic();
+            }
+            else if (typeof(T) == typeof(sbyte))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new SByteArithmetic();
+            }
+            else if (typeof(T) == typeof(short))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new ShortArithmetic();
+            }
+            else if (typeof(T) == typeof(uint))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new UIntArithmetic();
+            }
+            else if (typeof(T) == typeof(ulong))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new ULongArithmetic();
+            }
+            else if (typeof(T) == typeof(ushort))
+            {
+                return (IPrimitiveDataFrameColumnArithmetic<T>)new UShortArithmetic();
+            }
+            return null;
+        }
+    }
+
+    internal class BoolArithmetic : IPrimitiveDataFrameColumnArithmetic<bool>
+    {
+        public void Add(PrimitiveDataFrameColumnContainer<bool> left, PrimitiveDataFrameColumnContainer<bool> right)
```

Comment:
VS has a magic refactoring that will implement an interface for you. When it does this it uses NotImplementedEXception. You should probably use that.

NotImplementedException --> I am not done here
NotSupportedException --> It is not meaningful to call this (usually when something really should be abstract)
PlatformNotSupportedException --> Works on other platforms, not this one

Reply:
ie in theory in CoreFX there are zero NotImplementedException, but in practice I see we weren't consistent

Reply:
NotSupportedException is the right thing here I think. It doesn't make sense to call any arithmetic ops on a bool